### PR TITLE
[feat]: [CDS-76717]: Added build and build source type info in trigger activity page response

### PR DESCRIPTION
**Pull request by rathodmeetsatish**
## Describe your changes
Added build and build source type info in trigger activity page response
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51486)

### CODE REVIEW

**Comment by rathodmeetsatish** [comment]
trigger uts

**Comment by rathodmeetsatish** [comment]
trigger uts

File: pipeline-service/service/src/main/java/io/harness/pms/triggers/webhook/helpers/TriggerEventExecutionHelper.java [hunk 1]
```diff
@@ -308,8 +310,10 @@ public TriggerEventResponse generateEventHistoryForError(TriggerWebhookEvent tri
 
     TargetExecutionSummary targetExecutionSummary = TriggerEventResponseHelper.prepareTargetExecutionSummary(
         (PlanExecution) null, triggerDetails, runtimeInputYaml);
+
+    NGTriggerConfigV2 ngTriggerConfigV2 = ngTriggerElementMapper.toTriggerConfigV2(ngTriggerEntity);
```
Comment (vcalasansh) [inline pipeline-service/service/src/main/java/io/harness/pms/triggers/webhook/helpers/TriggerEventExecutionHelper.java:314]: Isn't `ngTriggerConfigV2` already inside `triggerDetails`?

**Review by vcalasansh — COMMENTED** [review]
Added a comment

**Comment by rathodmeetsatish** [comment]
trigger uts

**Review by vcalasansh — COMMENTED** [review]
> Reviewed 7 of 9 files at r1, 2 of 2 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51486)* status: 10 file(s) to review. 0 of 2 approvals obtained.

**Review by vcalasansh — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51486#-:-NbQt2Cm1Dj0WRM72w3V:bnfp4nl)* status: 10 file(s) to review. 1 of 2 approvals obtained.

**Review by vcalasansh — APPROVED** [review]

**Review by abhinav-harness — APPROVED** [review]

**Comment by rathodmeetsatish** [comment]
trigger review

**Pull request merged**